Thesis/StepIndexedRelBigStepTypedAnfIlcCorrect.agda

```diff
@@ -262,12 +317,25 @@ mutual
           k
       }
 
+rrelV3→⊕ : ∀ {τ n} v1 dv v2 → rrelV3 τ v1 dv v2 n → v1 ⊕ dv ≡ v2
+rrelV3→⊕ (closure {Γ} t ρ) (dclosure dt ρ₁ dρ) (closure t₁ ρ₂) ((refl , refl) , refl , p , refl , refl , dvv) with Γ ≟Ctx Γ
+rrelV3→⊕ (closure {Γ} t ρ) (dclosure .(derive-dterm t) .ρ dρ) (closure .t .(ρ ⊕ρ dρ)) ((refl , refl) , refl , refl , refl , refl , dvv) | yes refl = refl
+... | no ¬q = ⊥-elim (¬q refl)
+rrelV3→⊕ (natV n) (dnatV dn) (natV .(dn + n)) refl rewrite +-comm n dn = refl
```

Comment:
@yurug Here I show that ⊕ and validity agree. I hadn't noticed the issue with ⊕ and mismatched environments (which might also affect our definition of ⊕ in Coq?), that was tedious to fix.
As expected, I had to restrict validity to prove this, and adapting the proof of the fundamental lemma was not hard.